src/mono/mono/component/debugger-agent.c

```diff
@@ -125,6 +127,47 @@
 #pragma warning(disable:4312) // FIXME pointer cast to different size
 #endif
 
+#define GENERATE_TRY_GET_CLASS_WITH_CACHE_DBG(shortname,name_space,name) \
+MonoClass*	\
+mono_class_try_get_##shortname##_class (void)	\
+{	\
+	static volatile MonoClass *tmp_class;	\
+	static volatile gboolean inited;	\
+	MonoClass *klass = (MonoClass *)tmp_class;	\
+	mono_memory_barrier ();	\
+	if (!inited) {	\
+		klass = mono_class_try_load_from_name (get_mono_defaults ()->corlib, name_space, name);	\
+		tmp_class = klass;	\
+		mono_memory_barrier ();	\
+		inited = TRUE;	\
+	}	\
+	return klass;	\
+}
+
+#ifndef MONO_HANDLE_TRACK_OWNER
+
+#define MONO_HANDLE_NEW_DBG(type, object) \
+	(MONO_HANDLE_CAST_FOR (type) (mono_handle_new (MONO_HANDLE_TYPECHECK_FOR (type) (object), get_mono_thread_info_current_var ())))
```

Comment:
That `mono_thread_info_current_var` stuff is just a hack that was added to try and avoid calling `mono_thread_info_current()` too often (it works by making `HANDLE_FUNCTION_ENTER` make a local variable `mono_thread_info_current_var` that is assigned initialized to `mono_thread_info_current()`, and then it tries to use that value in `MONO_HANDLE_NEW`).  But it's just an optimization for frequently executed icalls.  You can just make `MONO_HANDLE_NEW_DBG` use `mono_thread_info_current()` directly.

Reply:
done.